src/reporting.js

```diff
@@ -366,7 +366,14 @@ async function generateReport(console, httpUrl, matrixFile, filePath, tempDir, s
     console.info(`Running command ${cmd}`);
     const result = await executeCommand(cmd);
 
-    reportCache[reportHash] = result;
+    const config = getConfig();
+
+    if (
+        !config.scan.doNotCacheExitCodes
```

Comment:
In the interest of being paranoid around JS, I think I would prefer to make this explicit.
What are you interested in here? Non-null/undefinedness? Emptiness?

This is apparently not a very Pythonic land:
```
> ![]
false
> [] == true
false
> [] == false
true
```

Reply:
I'm interested in all kind of false, i.e. I want this line to be `true` unless `config.scan.doNotCacheExitCodes` is a non-empty array, but I guess I can do these two checks separately.

Reply:
thanks, I think it's much more sane to have the code spell it out clearly as it does now (especially for non-JSers)